ql/src/java/org/apache/hadoop/hive/ql/ddl/table/metaref/AlterTableCreateMetaRefAnalyzer.java

```diff
@@ -16,92 +16,159 @@
+    AlterTableTypeReq alterTableTypeReq = new AlterTableTypeReq();
 
-    String branchName = command.getChild(0).getText();
+    String metaRefName = command.getChild(0).getText();
+    alterTableTypeReq.setTableName(tableName);
+    alterTableTypeReq.setMetaRefName(metaRefName);
     Long snapshotId = null;
     Long asOfTime = null;
     Long maxRefAgeMs = null;
     Integer minSnapshotsToKeep = null;
     Long maxSnapshotAgeMs = null;
+    AlterTableType alterTableType = command.getType()
+        == HiveParser.TOK_ALTERTABLE_CREATE_BRANCH ? AlterTableType.CREATE_BRANCH : AlterTableType.CREATE_TAG;
     for (int i = 1; i < command.getChildCount(); i++) {
       ASTNode childNode = (ASTNode) command.getChild(i);
       switch (childNode.getToken().getType()) {
       case HiveParser.TOK_AS_OF_VERSION:
         snapshotId = Long.parseLong(childNode.getChild(0).getText());
+        alterTableTypeReq.setSnapshotId(snapshotId);
         break;
       case HiveParser.TOK_AS_OF_TIME:
         ZoneId timeZone = SessionState.get() == null ? new HiveConf().getLocalTimeZone() :
             SessionState.get().getConf().getLocalTimeZone();
         TimestampTZ ts = TimestampTZUtil.parse(stripQuotes(childNode.getChild(0).getText()), timeZone);
         asOfTime = ts.toEpochMilli();
+        alterTableTypeReq.setAsOfTime(asOfTime);
         break;
       case HiveParser.TOK_RETAIN:
         String maxRefAge = childNode.getChild(0).getText();
         String timeUnitOfBranchRetain = childNode.getChild(1).getText();
-        maxRefAgeMs = TimeUnit.valueOf(timeUnitOfBranchRetain.toUpperCase(Locale.ENGLISH))
-            .toMillis(Long.parseLong(maxRefAge));
+        maxRefAgeMs =
+            TimeUnit.valueOf(timeUnitOfBranchRetain.toUpperCase(Locale.ENGLISH)).toMillis(Long.parseLong(maxRefAge));
+        alterTableTypeReq.setMaxRefAgeMs(maxRefAgeMs);
         break;
       case HiveParser.TOK_WITH_SNAPSHOT_RETENTION:
         minSnapshotsToKeep = Integer.valueOf(childNode.getChild(0).getText());
+        alterTableTypeReq.setMinSnapshotsToKeep(minSnapshotsToKeep);
         if (childNode.getChildren().size() > 1) {
           String maxSnapshotAge = childNode.getChild(1).getText();
           String timeUnitOfSnapshotsRetention = childNode.getChild(2).getText();
           maxSnapshotAgeMs = TimeUnit.valueOf(timeUnitOfSnapshotsRetention.toUpperCase(Locale.ENGLISH))
               .toMillis(Long.parseLong(maxSnapshotAge));
+          alterTableTypeReq.setMaxSnapshotAgeMs(maxSnapshotAgeMs);
         }
         break;
       default:
-        throw new SemanticException("Unrecognized token in ALTER CREATE BRANCH statement");
+        throw new SemanticException("Unrecognized token in ALTER " + alterTableType.getName() + " statement");
       }
     }
+    alterTableDesc = getAlterTableDesc(alterTableTypeReq);
+    rootTasks.add(TaskFactory.get(new DDLWork(getInputs(), getOutputs(), alterTableDesc)));
+  }
+
+  public class AlterTableTypeReq{
```

Comment:
could we mark it as a package-private?

Reply:
Removed the public setters. Make it protected as the child class `AlterTableCreateBranchAnalyzer `& `AlterTableCreateTagAnalyzer` will use it.